datatypes/src/plots/mod.rs

```diff
@@ -6,27 +6,40 @@ use serde::{Deserialize, Serialize};
 use std::fmt::Debug;
 
 pub trait Plot {
-    type PlotDataMetadataType: Debug + PartialEq + Serialize;
-
     /// Creates a Vega string for embedding it into a Html page
     ///
     /// # Errors
     ///
     /// This method fails on internal errors of the plot.
     ///
-    fn to_vega_embeddable(
-        &self,
-        allow_interactions: bool,
-    ) -> Result<PlotData<Self::PlotDataMetadataType>>;
+    fn to_vega_embeddable(&self, allow_interactions: bool) -> Result<PlotData>;
 
-    fn to_png(&self, width_px: u16, height_px: u16) -> Vec<u8>;
+    // TODO: create some PNG output, cf. https://github.com/procyon-rs/vega_lite_3.rs/issues/18
+    // fn to_png(&self, width_px: u16, height_px: u16) -> Vec<u8>;
 }
 
-#[derive(Debug, Deserialize, PartialEq, Serialize)]
-pub struct PlotData<M>
-where
-    M: Debug + PartialEq + Serialize,
-{
+#[derive(Debug, Clone, Deserialize, PartialEq, Serialize)]
+pub struct PlotData {
     pub vega_string: String,
-    pub metadata: M,
+    pub metadata: PlotMetaData,
+}
+
+#[derive(Debug, Clone, Deserialize, PartialEq, Serialize)]
+#[serde(untagged)]
+pub enum PlotMetaData {
```

Comment:
What is the scope of this? Right now you use it for specifying whether (and how) the plot is interactable, but I think this data type is not really extendable, is it? What if there are multiple different interaction types in the future or other meta data like additional text to display. Ithink meta data should be a struct and the interaction a field. But we can also tackle this later on.

Reply:
I don't know, you mean it should just be a box of some JSON Value?

Reply:
No, but rather a struct because it can be more easily extended.
````
pub struct PlotMetaData {
   is_interactive: bool,
````

Reply:
But then you don't get the name of the interaction component that you need for accessing it from the vega plot.

Reply:
then rename `PlotMetaData` to `InteractionType` and make
```
pub struct PlotMetaData {
   interaction: InteractionType,
```

but like I said we can also do this later on. maybe the interaction will be the only plot meta data we will ever need.

Reply:
I would vote for this. We change it once we have the second type and the current implementation seems odd.